File: ProjC1/src/projc1/Game/PositionGenerator.java

package projc1.Game;

class PositionGenerator {
    private java.util.Random random;
    
    public PositionGenerator() {
        this.random = new java.util.Random();
    }
    
    // DEPENDENCIA: Este método depende de Position pero no la almacena
    public Position generateRandomPosition(int maxX, int maxY, Position playerPos, int minDistance) {
        int x, y;
        Position newPos;
        
        do {
            x = random.nextInt(maxX);
            y = random.nextInt(maxY);
            newPos = new Position(x, y);
        } while (newPos.distanceTo(playerPos) < minDistance);
        
        return newPos;
    }
}

File: ProcC1_pyver/game/components/position_generator.py
import random
from game.components.position import Position

class PositionGenerator:
    def __init__(self):
        self.random = random.Random()
    
    def generate_random_position(self, max_x, max_y, player_pos, min_distance):
        x, y = 0, 0
        new_pos = None
        
        for _ in range(100): 
            x = self.random.randint(20, max_x - 20)
            y = self.random.randint(20, max_y - 20)
            new_pos = Position(x, y)
            
            if new_pos.distance_to(player_pos) >= min_distance:
                break
        
        return new_pos